src/pages/node-express/lesson11-a1.mdx

```diff
@@ -0,0 +1,948 @@
+  password2 = document.getElementById("password2");
+  const registerButton = document.getElementById("register-button");
+  const registerCancel = document.getElementById("register-cancel");
+
+  registerDiv.addEventListener("click", async (e) => {
+    if (inputEnabled && e.target.nodeName === "BUTTON") {
+      if (e.target === registerButton) {
+        if (password1.value != password2.value) {
+          message.textContent = "The passwords entered do not match.";
+        } else {
+          enableInput(false);
+
+          try {
+            const response = await fetch("/api/v1/auth/register", {
+              method: "POST",
+              headers: {
+                "Content-Type": "application/json",
+              },
+              body: JSON.stringify({
+                name: name.value,
+                email: email1.value,
+                password: password1.value,
+              }),
+            });
+
+            const data = await response.json();
+            if (response.status === 201) {
+              message.textContent = `Registration successful.  Welcome ${data.user.name}`;
+              setToken(data.token);
+
+              name.value = "";
+              email1.value = "";
+              password1.value = "";
+              password2.value = "";
+
+              showJobs();
+            } else {
+              message.textContent = data.msg;
+            }
+          } catch (err) {
+            console.error(err);
+            message.textContent = "A communications error occurred.";
+          }
+
+          enableInput(true);
+        }
+      } else if (e.target === registerCancel) {
+        name.value = "";
+        email1.value = "";
+        password1.value = "";
+        password2.value = "";
+        showLoginRegister();
+      }
+    }
+  });
+};
+
+export const showRegister = () => {
+  email1.value = null;
+  password1.value = null;
+  password2.value = null;
+  setDiv(registerDiv);
+};
+```
+
+Notice that we've now made the click event listener callback function `async`.
+We then check to see if the user entered matching passwords in the two inputs. If they did,
+disable clicking on the buttons and then make the `fetch()` call to the register endpoint.
+
+If the call is successful, we parse the data from the response, display a message to the user,
+and save the token to local storage using the function exported from `index.js`. We can then display the jobs page.
+If the call was not successful we will continue to show the register page and will dipslay the error message from the
+response to the user. If any errors occur, then we catch them and continue eto show the register page
+and show a generic error message to the user.
+
+After we're done processing the call (whether it was successful or not) we re-enable clicking on the buttons.
+
+Notice that we always clear out the input values before we switch to another page. We do not want those
+to live on in memory.
```

Comment:
Added a little more in-depth explanation of the changes made here; with the aim of making the following examples easier to follow along.